p2pv2/stream_limit.go

```diff
@@ -0,0 +1,117 @@
+package p2pv2
+
+import (
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	"github.com/patrickmn/go-cache"
+
+	"github.com/xuperchain/log15"
+)
+
+// StreamLimit limit the peerID amount of same ip
+type StreamLimit struct {
+	// Store all streams available
+	// key: addr value: peerID
+	streams *cache.Cache
```

Comment:
why use cache, not map ?

Reply:
You can use map, sync.Map or cache.Cache, No much difference.

Reply:
> why use cache, not map ?

I have corrected the issue using sync.Map instead of cache.Cache.